src/Admin/Form/RoleType.php

```diff
@@ -14,15 +16,39 @@
 
 class RoleType extends AbstractType
 {
+    private PluginService $pluginService;
+
+    public function __construct(PluginService $pluginService)
+    {
+        $this->pluginService = $pluginService;
+    }
+
     public function configureOptions(OptionsResolver $resolver): void
     {
         $resolver->setDefaults([
             'data_class' => Role::class,
+            'permissions' => [],
         ]);
     }
 
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
+        $forumifyPermissions = ForumifyBundle::getPermissions();
+        $installedPlugins = $this->pluginService->getInstalledPlugins();
```

Comment:
instead of getting the installed plugins, can't you use `PluginRepository`, use `findByActive()`, and then loop over the plugins and get the plugin class by doing `$plugin->getPlugin()`?

Then in the loop check if instance of `AbstractForumifyPlugin` (to filter out theme plugins).

Should make the code a lot simpler and it'd filter out inactive plugins. (if the plugin isn't active, its permissions aren't used anyway)